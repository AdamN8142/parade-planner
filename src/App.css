

/* .entire-page {
  background-color:wheat}  */
