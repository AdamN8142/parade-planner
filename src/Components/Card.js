import React, { Component } from 'react';

class Card extends Component {
  constructor() {
    super();
  }

  render() {
    return(
      <div>
        <article>{this.props.name}</article>
      </div>
      
    )
  }
}


export default Card;