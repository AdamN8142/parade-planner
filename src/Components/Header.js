import React from 'react';

const Header = () => {
  return (
    <div className='Header'>
      <h1>PARADE PLANNER</h1>
      <h2>ULTIMATE MARDI GRAS EDITION</h2>
    </div>
  )
}

export default Header;
