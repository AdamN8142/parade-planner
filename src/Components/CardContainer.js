import React, { Component } from 'react';
import Card from './Card'

class CardContainer extends Component {
  constructor(){
      super();
  }

    render() {
      return(
          <div>
              <Card />
          </div>
      )
  }     
}



export default CardContainer;