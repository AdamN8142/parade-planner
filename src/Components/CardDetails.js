import React, { Component } from 'react';
import Restaurant from './Restaurant';
import "../Card.css";


class CardDetails extends Component {
  constructor() {
    super();
  }
  

  render() {
    // const displayCard = this.props.parades.filter((parade) => {
    //   return parade.id == this.props.id
    // })
    // console.log(displayCard, "heyy")
    const{parade,restaurants,toggle} = this.props
    const sortRestaurants = restaurants.filter((res) => {
      return res.location == parade.location
    })
        
    return(
      <div className = "pop-up">
        <button onClick={toggle}>X</button>
        <h1>{parade.name}</h1>
        <img src={parade.img}/>
        <p>location: {parade.location}</p>
        <p>{parade.date} | {parade.startTime}</p>
        <p>Number of Floats: {parade.numberOfFloats}</p>
        {/* <div className="> */}
          <div className="restaurant-container">
              <Restaurant sortedRestaurants={sortRestaurants}/>
          </div>
        {/* </div> */}
      </div>
    )
  }
}
        


      


export default CardDetails;