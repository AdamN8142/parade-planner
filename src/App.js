import React, { Component } from 'react';
import './App.css';
import CardContainer from './CardContainer';
import FilterContainer from './FilterContainer';

class App extends Component {
  render() {
    return (
      <div className="App">
      <CardContainer />
      <FilterContainer />
      </div>
    );
  }
}

export default App;
