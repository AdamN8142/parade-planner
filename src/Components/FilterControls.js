import React, { Component } from 'react';
import CardContainer from './CardContainer';
// import '../FilterControls.scss';

class FilterControls extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: '',
      dayOrNight: '',
      date: ''
    }
  }

  getClickedValue = (event) => {
    this.setState({
      [event.target.id]: event.target.value
    }, this.filterParades)
  }

  filterParades = () => { 
    let keyArr = ['location', 'dayOrNight', 'date']
    let filteredParades = this.props.parades.filter(parade => {
      return keyArr.every(key => {
        return this.state[key] === parade[key] || this.state[key] === ''
      })
    })
    return filteredParades
  }

  resetFilter = () => {
    this.setState({
      location: '',
      dayOrNight: '',
      date: ''
    })
  }

  render() {
    return (
      <div>
        <div className ="filters-section">
              <div className="filters">
                <select className="filter" id="location" onChange={this.getClickedValue}>
                  <option value="">--Select Location--</option>
                  <option value="Metairie">Metairie</option>
                  <option value="Uptown New Orleans">Uptown New Orleans</option>
                </select>
                <select className="filter" id="dayOrNight" onChange={this.getClickedValue}>
                  <option value="">--Select Time--</option>
                  <option value="am">AM</option>
                  <option value="pm">PM</option>
                </select>
              <select className="filter" id="date" onChange={this.getClickedValue}>
                <option value="">--Select Date--</option>
                {this.props.parades.reduce((acc, currentParade) => {
                  if(!acc.includes(currentParade.date)){acc.push(currentParade.date)}
                    return acc
                  },[]).map((name)=> {
                    return (<option value={[name]}>{[name]}</option>)
                  })
                }
              </select>
              <div className="buttons">
              {/* <button className="filter-button"onClick={this.showAllParades}>Show All</button> */}
              <button className="filter-button"onClick={this.resetFilter}>RESET</button>
            </div>
            </div>
        </div>
        <CardContainer 
        filteredParades={this.filterParades()}
        parades={this.props.parades}
        restaurants={this.props.restaurants}

        />
      </div>
    )
  }
}

export default FilterControls;

