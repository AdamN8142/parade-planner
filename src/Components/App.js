import React, { Component } from 'react';
import '../App.css';
import FilterControls from './FilterControls';
import Header from './Header';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      parades : [],
      restaurants : [],
    }
  }
  

  getData = (request) => {
    
    fetch(`http://whateverly-datasets.herokuapp.com/api/v1/${request}`)
    .then(response => response.json())
    .then(data => {
        this.setState({
          [request]: data[request]
        })
      })
      .catch(error => {
        throw new Error(error)
      });
  }



  componentDidMount() {
    this.getData('restaurants')
    this.getData('parades')
  }
  
  render() {
    return (
      <div className="entire-page">
        <Header />
        <FilterControls 
        parades={this.state.parades}
        restaurants={this.state.restaurants}
        />
      </div>
    );
  }
}

export default App;
