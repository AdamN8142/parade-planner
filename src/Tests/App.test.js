// App component does not take in any props, so we do not need to create any mock data. 
// The other components did need mock data because the components needed props.

import React from 'react';
import ReactDOM from 'react-dom';
import {App} from '../Components/App';
import { shallow } from 'enzyme';


// Do we need mock data here?
const mockData = [
	{
		"name": "Krewe of Little Rascals",
		"location": "Metairie",
		"dayOrNight": "pm",
		"startTime": "12:00pm",
		"date": "2/17/2019",
		"themes": ["children", "cars", "family"],
		"numberOfFloats": 85,
		"id": 1,
		"img": "https://assets.simpleviewcms.com/simpleview/image/fetch/c_limit,q_75,w_1200/https://assets.simpleviewinc.com/simpleview/image/upload/crm/jefferson/37872354-littlerascals2-jpg0_cc32794e-5056-a36a-08469bbb1ddece75.jpg"
	},
	{
		"name": "Krewe of Excalibur",
		"location": "Uptown New Orleans",
		"dayOrNight": "pm",
		"startTime": "7:00pm",
		"date": "2/22/2019",
		"themes": ["trucks", "floats", "medieval", "magic"],
		"numberOfFloats": 112,
		"id": 2,
		"img": "https://www.hlimg.com/images/events/738X538/25mardi6502_1546063931e.jpg"
	}
]

describe('App', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
	});
	
	it('should render without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<App/>, div);
		ReactDOM.unmountComponentAtNode(div);
	})

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({ parades: [], restaurants: [] });
	});
	

  // it('should get update state when getData is called', () => {
	// 	expect(wrapper.state()).toEqual({ parades: [], restaurants: [] });
	// 	fetch('http://whateverly-datasets.herokuapp.com/api/v1/restaurants') = jest.fn()
	// 	getData(restaurants);
		
	// });
	
	it('should call getData with correct parameters', () => {
		const getData = jest.fn()
		wrapper.instance().getData('restaurants')
		expect(getData).toHaveBeenCalled(1)
	})
});