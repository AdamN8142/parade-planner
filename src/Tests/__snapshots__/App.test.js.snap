// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="entire-page"
>
  <Header />
  <FilterControls
    parades={Array []}
    restaurants={Array []}
  />
</div>
`;
